c_glib/test/test-in-memory-scan-task.rb

```diff
@@ -0,0 +1,50 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+class TestInMemoryScanTask < Test::Unit::TestCase
+  include Helper::Buildable
+
+  def setup
+    omit("Arrow Dataset is required") unless defined?(ArrowDataset)
+    fields = [
+      Arrow::Field.new("visible", Arrow::BooleanDataType.new),
+      Arrow::Field.new("point", Arrow::Int32DataType.new),
+    ]
+    @schema = Arrow::Schema.new(fields)
+    @record_batches = [
+      [
+        build_boolean_array([true, false, true]),
+        build_int32_array([1, 2, 3]),
+      ],
+      [
+        build_boolean_array([false, true, false, true]),
+        build_int32_array([-1, -2, -3, -4]),
+      ]
+    ].collect do |columns|
+      Arrow::RecordBatch.new(@schema, columns[0].length, columns)
+    end
+  end
+
+  def test_execute
+    @scan_task = ArrowDataset::InMemoryScanTask.new(@record_batches,
+                                                    ArrowDataset::ScanOptions.new(@schema),
+                                                    ArrowDataset::ScanContext.new)
+    p ArrowDataset::ScanTask.instance_methods.sort
+    assert_equal(@record_batches,
+                 @scan_task.execute.to_a)
```

Comment:
@kou I got the following error here:

```
     45:                                                     ArrowDataset::ScanContext.new)
     46:     p ArrowDataset::ScanTask.instance_methods.sort
     47:     assert_equal(@record_batches,
  => 48:                  @scan_task.execute.to_a)
     49:   end
     50: end
/Users/mrkn/src/github.com/apache/arrow/c_glib/test/test-in-memory-scan-task.rb:48:in `test_execute'
Error: test_execute(TestInMemoryScanTask): NoMethodError: undefined method `execute' for #<ArrowDataset::InMemoryScanTask:0x00007ff506b570e0>
```

Do you know why there is not `ArrowDataset::ScanTask#execute` instance method?

Reply:
Could you remove `:` after `(nullable)` in `Returns: (nullable): (transfer full)` of `gad_scan_task_execute()`?

```diff
diff --git a/c_glib/arrow-dataset-glib/scanner.cpp b/c_glib/arrow-dataset-glib/scanner.cpp
index 792366dea..e4a204fad 100644
--- a/c_glib/arrow-dataset-glib/scanner.cpp
+++ b/c_glib/arrow-dataset-glib/scanner.cpp
@@ -420,7 +420,7 @@ gad_scan_task_class_init(GADScanTaskClass *klass)
  * @scan_task: A #GADScanTask.
  * @error: (nullable): REturn location for a #GError or %NULL.
  *
- * Returns: (nullable): (transfer full): A newly created #GArrowRecordBatchIterator,
+ * Returns: (nullable) (transfer full): A newly created #GArrowRecordBatchIterator,
  *   or %NULL on error.
  *
  * Since: 1.0.0
```

Reply:
Resolved!  Thank you very much!!